liquidity-amo/contracts/V3AMO.sol

```diff
@@ -0,0 +1,578 @@
+     * @return boostRemoved Amount of BOOST tokens removed from the pool.
+     * @return usdRemoved Amount of USD tokens removed from the pool.
+     * @return usdAmountIn Amount of USD tokens swapped into BOOST.
+     * @return boostAmountOut Amount of BOOST tokens bought and burned.
+     */
+    function _unfarmBuyBurn(
+        uint256 liquidity,
+        uint256 minBoostRemove,
+        uint256 minUsdRemove
+    )
+        internal
+        override
+        returns (uint256 boostRemoved, uint256 usdRemoved, uint256 usdAmountIn, uint256 boostAmountOut)
+    {
+        // Step 1: Remove liquidity from the pool
+        // Remove liquidity and store the amounts of USD and BOOST tokens received
+        uint256 amount0FromBurn;
+        uint256 amount1FromBurn;
+        if (poolType == PoolType.ALGEBRA_INTEGRAL)
+            (amount0FromBurn, amount1FromBurn) = IAlgebraIntegralPool(pool).burn(
+                tickLower,
+                tickUpper,
+                uint128(liquidity),
+                ""
+            );
+        else (amount0FromBurn, amount1FromBurn) = IUniswapV3Pool(pool).burn(tickLower, tickUpper, uint128(liquidity));
+        (boostRemoved, usdRemoved) = sortAmounts(amount0FromBurn, amount1FromBurn);
+
+        // Ensure the BOOST amount removed from our full-range position is greater than or equal to the USD amount removed
+        if (boostRemoved < minBoostRemove) revert InsufficientOutputAmount(boostRemoved, minBoostRemove);
+        if (usdRemoved < minUsdRemove) revert InsufficientOutputAmount(usdRemoved, minUsdRemove);
+
+        if (poolType == PoolType.SOLIDLY_V3) {
+            address feeCollector = ISolidlyV3Factory(ISolidlyV3Pool(pool).factory()).feeCollector();
+            IRewardsDistributor(feeCollector).collectPoolFees(pool);
+        }
+        uint128 amount0Collected;
+        uint128 amount1Collected;
+        (amount0Collected, amount1Collected) = IUniswapV3Pool(pool).collect(
+            address(this),
+            tickLower,
+            tickUpper,
+            type(uint128).max,
+            type(uint128).max
+        );
+        (uint256 boostCollected, uint256 usdCollected) = sortAmounts(amount0Collected, amount1Collected);
+
+        // Ensure the BOOST amount removed from our full-range position is greater than or equal to the USD amount removed
+        // this calculation/check is valid because based on our full-range liquidity (not on the aggregate pool liquidity)
+        if ((boostRemoved * validRemovingRatio) / FACTOR < toBoostAmount(usdRemoved))
+            revert InvalidRatioToRemoveLiquidity();
+
+        // Step 4: Use quoteSwap to determine the USD needed to bring the price back to peg
+        (int256 amount0, int256 amount1) = IUniswapV3Pool(pool).swap(
+            address(this),
+            boost > usd, // zeroForOne
+            int256(usdRemoved), // Maximum USD to use for the swap
```

Comment:
Should we use `usdCollected` instead of `usdRemoved` both here and on L400? Since we may also receive additional usd tokens as fees.

Reply:
No, it's intentional. We actually burn the BOOST portion of the fees (on [line 404](https://github.com/sherlock-audit/2024-10-axion/blob/affa6528fa7548df8d914ab9c04bba909e1d62ec/liquidity-amo/contracts/V3AMO.sol#L404)) and keep the USD portion in the contract so that we can withdraw it later.